src/scripts/main.js

```diff
@@ -1,3 +1,16 @@
 'use strict';
 
-// write code here
+const wall = document.querySelector('.wall');
+const spider = document.querySelector('.spider');
+
+const wallWidth = wall.offsetWidth;
+const wallHeight = wall.offsetHeight;
+const spiderWidth = spider.offsetWidth;
+const spiderHeight = spider.offsetHeight;
+
+const centerX = (wallWidth - spiderWidth) / 2;
+const centerY = (wallHeight - spiderHeight) / 2;
+
+spider.style.position = 'absolute';
```

Comment:
The 'absolute' position is used here. Make sure that the parent element of the spider has a position other than 'static' (the default), otherwise, it might not behave as expected. It's a good practice to ensure this in your CSS.